[Bug]: error TS2305: Module '"yargs-parser"' has no exported member XXX

### Version

28.0.7

### Steps to reproduce

npm install jest &&  ts-jest && tsc


### Expected behavior

type script compile should pass

### Actual behavior

type script compile failed

### Debug log

09:05:32  node_modules/@types/yargs/index.d.ts(28,10): error TS2305: Module '"yargs-parser"' has no exported member 'DetailedArguments'.
09:05:32  node_modules/@types/yargs/index.d.ts(28,29): error TS2305: Module '"yargs-parser"' has no exported member 'Configuration'.

### Additional context

https://github.com/yargs/yargs-parser/issues/452

### Environment

```shell
Ubuntu 18.04.6 LTS (GNU/Linux 5.4.0-1078-aws x86_64)
```

I am not getting an error on npm install, but same error at compile time

`node_modules/@types/yargs/index.d.ts:28:10 - error TS2305: Module '"yargs-parser"' has no exported member 'DetailedArguments'.

28 import { DetailedArguments, Configuration } from 'yargs-parser';
            ~~~~~~~~~~~~~~~~~

node_modules/@types/yargs/index.d.ts:28:29 - error TS2305: Module '"yargs-parser"' has no exported member 'Configuration'.

28 import { DetailedArguments, Configuration } from 'yargs-parser';`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,12 @@
+## [28.0.8](https://github.com/kulshekhar/ts-jest/compare/v28.0.7...v28.0.8) (2022-08-14)
+
+
+### Bug Fixes
+
+* allow `.mts` to be processed ([#3713](https://github.com/kulshekhar/ts-jest/issues/3713)) ([effae71](https://github.com/kulshekhar/ts-jest/commit/effae717369860e16cb0ccbf24027651493b9bf1)), closes [#3702](https://github.com/kulshekhar/ts-jest/issues/3702)
+
+
+
 ## [28.0.7](https://github.com/kulshekhar/ts-jest/compare/v28.0.6...v28.0.7) (2022-07-15)
 
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ts-jest",
-  "version": "28.0.7",
+  "version": "28.0.8",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "bin": {

--- a/website/docs/guides/esm-support.md
+++ b/website/docs/guides/esm-support.md
@@ -93,3 +93,22 @@ module.exports = {
   }
 }
 ```
+
+#### Support `.mts` extension
+
+To work with `.mts` extension, besides the requirement to run Jest and `ts-jest` in ESM mode, there are a few extra requirements to be met:
+
+- `package.json` should contain `"type": "module"`
+- A custom Jest resolver to resolve `.mjs` extension, see our simple one at https://github.com/kulshekhar/ts-jest/blob/main/e2e/native-esm-ts/mjs-resolver.ts
+- `tsconfig.json` should at least contain these following options
+
+```json
+{
+  "compilerOptions": {
+    "module": "Node16", // or "NodeNext"
+    "target": "ESNext",
+    "moduleResolution": "Node16", // or "NodeNext"
+    "esModuleInterop": true
+  }
+}
+```

--- a/website/versioned_docs/version-28.0/guides/esm-support.md
+++ b/website/versioned_docs/version-28.0/guides/esm-support.md
@@ -93,3 +93,22 @@ module.exports = {
   }
 }
 ```
+
+#### Support `.mts` extension
+
+To work with `.mts` extension, besides the requirement to run Jest and `ts-jest` in ESM mode, there are a few extra requirements to be met:
+
+- `package.json` should contain `"type": "module"`
+- A custom Jest resolver to resolve `.mjs` extension, see our simple one at https://github.com/kulshekhar/ts-jest/blob/main/e2e/native-esm-ts/mjs-resolver.ts
+- `tsconfig.json` should at least contain these following options
+
+```json
+{
+  "compilerOptions": {
+    "module": "Node16", // or "NodeNext"
+    "target": "ESNext",
+    "moduleResolution": "Node16", // or "NodeNext"
+    "esModuleInterop": true
+  }
+}
+```

